### ISSUE TITLE
Future of gwtbootstrap

### ISSUE THREAD

**Issue by ArloL** [gwtbootstrap3/gwtbootstrap3#281]
GWT Bootstrap provides a bunch of Widgets for using Bootstrap in GWT Projects. In the meantime GWT is moving towards deprecating Widgets. For example in Julien Dramaix's talk about [Best development practices for GWT web applications](https://drive.google.com/file/d/0BwVGJUurq6uVRXVQR1o4MHdnVk0/view) his first point is to avoid Widgets as much as possible. He actually uses GWT Bootstrap as an example.

There is talk about removing Widgets from GWT core for 3.0 and the consensus of GWT.create in Munich seemed to me that using HTML is the best way to create GWT applications.

Maybe we should move towards a light-weight wrapper around bootstrap using JsInterop and dropping the widgets. This would be something for a 2.0 release synced with GWT 3.0.

What do you think?

**Comment by remarkNS** [comment]
I agree with the statement above. If you take a look this project http://cleangwt.appspot.com/ you will see that the author (not me) is using idea exactly what you wrote above. I did some experiment where I take html from here http://bootsnipp.com/snippets/featured/parallax-login-form and map this to username, password and button which are GwtBootstrap3 widgets (my experiment was to map it to this library instead of cleangwt). It works really great. The point is that you do not have to create widgets in UIBinder way which mimics bootstrap, you can use it as is (just look html of paralax from link above, the only thing you need to do is to put id = "someID" on controls that you want to use in your controller). Even this http://www.bootstrapzero.com/bootstrap-template/facebook works great.

**Comment by jgodi** [comment]
I do like this concept. I think this will make gwtbootstrap3 very light and easier to maintain. Big, big things are coming in the future of GWT.

With these potential changes, we will have to leverage the community and you in making sure that it will be extremely solid and easy to use.

Thanks for bringing this up @ArloL.

**Comment by amitmun** [comment]
No doubt this is the right way to go. And I'm happy you guys are taking this direction which will make sure GwtBootstrap3 stays relevant in the future of GWT 3.0.
I've been working with GWT since 2007, and I feel that finally the web standards are getting solid enough to allow GWT take advantage of them and allow all the goodness in Java together with the power of the web (as opposed to GWT hiding the web like it was in GWT's early days).
My approach these days is: Let me work with the standard tools of the web (HTML, CSS and the more advanced tools) and then use the power of GWT to glue everything together, handle events and all the complex logic, talk to the server and pack it all in the most optimized way possible.

**Comment by amitmun** [comment]
A good approach would be to write something that can take any bootstrap files, parse them and create a Java GWT wrapper using JSInterop, the same way GWT-Polymer works (https://drive.google.com/a/mun.co.il/file/d/0B1YQojwwF_5xSUY2dV9Uc1IxRHc/view).
This will make sure anyone can work with their desired Bootstrap version and that will keep you up-to-date anytime. gwtbootstrap will also make sure to inject Bootstrap files and add any special glue required to make it super easy to use.
Widgets will be deprecated soon, and the race of continuously updating gwtbootstrap with new Bootstrap features, icons and styles can be avoided.

**Comment by CodeLessSolutions** [comment]
We are using gwtbootstrap3 this way: https://www.youtube.com/watch?v=ROZ5Oa6DoUg
Note how it is done without generating, compiling or redeploying anything.

**Comment by pellyadolfo** [comment]
I was a userof previous GWTBootstrap library (https://gwtbootstrap.github.io/) and I always thought that the value was on widgets BUT also that they are too tied to the Bootstrap version and therefore expensive to maintain. 

One question then. By removing widgets from GWTBootstrap3, what would be the added value of GWTBootstrap3 vs plain GWT+bootstrap3 files? Just curious.
